### PR TITLE
target: create an error when multiple tasks have the same target

### LINKED ISSUE #211: target: create an error when multiple tasks have the same target
Multiple tasks with the same target could lead to unexpected behavior.

### PR DESCRIPTION
**Pull request by zuzuleinen**
closes #211

### CODE REVIEW

File: bobtask/map.go [hunk 1]
```diff
@@ -194,3 +195,30 @@ func (tm Map) IgnoreChildTargets() (err error) {
...
+
+	for taskName, v := range tm {
+		target, _ := v.Target()
+		if target == nil {
+			continue
+		}
+		for _, t := range target.DockerImages() {
+			targetToTasks[t] = append(targetToTasks[t], taskName)
+		}
+		for _, t := range target.FilesystemEntriesRaw() {
```
Comment (Equanox) [inline bobtask/map.go:211]: When imports are also considered you should use `FilesystemEntries()` this let's you compare paths  in relation to the umbrella Bobfile.

**Review by Equanox — CHANGES_REQUESTED** [review]
The duplicate check for targets should be done after aggregation. An umbrella Bobfile can define targets inside the filesystem scope of a child Bobfile. Please also add a test for this case.

File: bobtask/map.go (hunk 1 above)
Comment (zuzuleinen) [reply bobtask/map.go:211]: If tasks from bobfiles cannot have targets from a above directories, do we really need to do this?

Comment (Equanox) [reply bobtask/map.go:211]: But tasks from umbrella Bobfiles can have targets in downstream directories.

**Pull request merged**